### DISCUSSION TITLE
Avoid hydration mismatch with viewport width

### DISCUSSION THREAD

**Discussion by mrc-bsllt** [nuxt/nuxt#28444] [Questions]
Hello everyone,
I need some advice, I sometimes have to render a different component depending on the size of the viewport. 

e.g.
``` javascript
<template>
  <div>
    <MobileComponent v-if="width < 1024" />
    <DesktopComponent v-else />
  </div>
</template>

<script setup lang="ts">
import { useWindowSize } from '@vueuse/core';
import MobileComponent from './components/MobileComponent.vue';
import DesktopComponent from './components/DesktopComponent.vue';

const { width } = useWindowSize();
</script>
```

In a case like this, if the viewport is already larger than 1024px, there is no problem, because between server and client the template remains the same. But of course, if I reload the page with a size smaller than 1024px, the hydration mismatch warning appears in the console. Do you have any suggestions for me other than to use the ClientOnly component?
Thank you all!

**Comment by tobijafischer** [comment]
Of course, there is always the option of using the `<ClientOnly>` [component](https://nuxt.com/docs/api/components/client-only), but for cases where this is not an option, I also don't always have a satisfactory solution. Would love to hear other ideas / workarounds.
